### ISSUE TITLE
Datatable destroy() (v1.10) and fnDestroy(v1.9) generates Detached Nodes and memory leaks

### ISSUE THREAD

**Issue by gicaz** [DataTables/DataTables#878]
Destroying and recreating tables in succession, generates memory leaks.
This is caused  by In https://github.com/DataTables/DataTables/blob/master/media/js/jquery.dataTables.js#L9326 (v1.10)
missing a for cycle to set "null" all settings attributes. 

for (var sKey in settings) {
settings[sKey] = null;
}

This problem there is in fnDestroy (v1.9) too. 
An exhaustive problem's analysis: http://www.mozartrocks.ro/dt-test/patch-1.html.

I hope you'll fix this quickly.
Thanks

**Comment by DataTables** [comment]
Can you give me a link to a page that demonstrates the issue please. If there are events not being tied or DOM elements not being realised, I'd like to understand why that is happening.

**Comment by gicaz** [comment]
At the moment, I cannot show the issue on a public link. I could send you a  timeline of the memory with and without fix. 
I found the bug performing a lot of consecutive destroy and reinitialize commands  querying a heavy dataset with over 8000 rows. This bug doesn't appear with lighter datasets or fewer requests. 
I tried your live example taken from the URL http://live.datatables.net/ too by executing the $('#example').DataTable().destroy(true)  and then the  "Run with js" commands a number of times. I attached the nodes' curve and the number of detatched nodes (opening the list of detached nodes the page crashed). 
![live datatable detached nodes](https://cloud.githubusercontent.com/assets/14332186/19595950/85b437aa-978c-11e6-9e0a-77cb68149e04.png)
![live datatable nodes curve](https://cloud.githubusercontent.com/assets/14332186/19595951/85b59c30-978c-11e6-942b-c795b58a22cd.png)
